effect/shared/src/main/scala/scalaz/effect/AsyncReturn.scala

```diff
@@ -9,22 +9,14 @@ package scalaz.effect
  * asynchronous action, `now` which represents a synchronously computed value,
  * or `maybeLater`, which represents an interruptible asynchronous action.
  */
-sealed abstract class AsyncReturn[A] { self =>
-  import AsyncReturn._
-
-  final def fold[Z](later: => Z, now: A => Z, maybeLater: Canceler => Z): Z =
-    self match {
-      case Now(v)        => now(v)
-      case MaybeLater(c) => maybeLater(c)
-      case _             => later
-    }
-}
+sealed abstract class AsyncReturn[E, A]
 object AsyncReturn {
-  type Canceler = Throwable => Unit
+  type Interruptor = Throwable => Unit
 
-  private final case object Later                    extends AsyncReturn[Nothing]
-  final case class Now[A](value: A)                  extends AsyncReturn[A]
-  final case class MaybeLater[A](canceler: Canceler) extends AsyncReturn[A]
+  private final case object Later extends AsyncReturn[Nothing, Nothing]
```

Comment:
There is also @TomasMikula's way of encoding these polymorphic values, which I have meant to deploy ever since I learned about it (see also: `IO.never`, `IO.unit`, and several other values which could benefit from this transformation).

I'll be sure I clean all this stuff up in a subsequent commit!